android-lib/src/main/java/com/kinvey/android/Client.kt

```diff
@@ -0,0 +1,1136 @@
+    //    private ConcurrentHashMap<String, AsyncCustomEndpoints> customeEndpointsCache;
+    private val customEndpoints: AsyncCustomEndpoints<*, *>? = null
+    private var pushProvider: AbstractPush? = null
+    private var userDiscovery: AsyncUserDiscovery? = null
+    private var userGroup: AsyncUserGroup? = null
+    private var clientUser: ClientUser? = null
+    //    private AsyncUser currentUser;
+    /**
+     * How long, in milliseconds, should offline wait before retrying a failed request
+     *
+     * @return the number of milliseconds for offline sync to wait between failures
+     */
+    var syncRate: Long = 0
+        private set
+    var batchRate: Long = 0
+        private set
+    var numberThreadPool: Int = 0
+        private set
+    var batchSize: Int = 0
+        private set
+    var isClientRequestMultithreading: Boolean = false
+        private set
+    private var cacheManager: RealmCacheManager? = null
+    var pushServiceClass: Class<*>? = null
+
+    init {
+        Logger.init(AndroidLogger())
+        sharedInstance = this as Client<User>
+        this.context = context
+        cacheManager = RealmCacheManager(encryptionKey, this)
+        syncCacheManager = RealmCacheManager(encryptionKey, "sync_", this)
+        userCacheManager = RealmCacheManager(encryptionKey, this)
+    }
+
+
+    @Deprecated("Renamed to {@link #setActiveUser(T)}")
+    fun setUser(user: T) {
+        synchronized(lock) {
+            this.user = user
+        }
+    }
+
+    override fun setActiveUser(user: T?) {
+        synchronized(lock) {
+            this.user = user
+        }
+    }
+
+    override fun getActiveUser(): T? {
+        synchronized(lock) {
+            return this.user
+        }
+    }
+
+    override fun performLockDown() {
+        //clear data cache and file cache
+        cacheManager?.clear()
+        //clear sync cache
+        syncCacheManager?.clear()
+        //clear user info cache
+        userCacheManager?.clear()
+        cacheManager = RealmCacheManager(encryptionKey, this)
+        syncCacheManager = RealmCacheManager(encryptionKey, "sync_", this)
+        userCacheManager = RealmCacheManager(encryptionKey, this)
+        val extensions = extensions
+        for (e in extensions) {
+            e.performLockdown(activeUser?.id)
+        }
+    }
+
+    /**
+     * Custom Endpoints factory method
+     *
+     *
+     * Returns the instance of [com.kinvey.java.CustomEndpoints] used for executing RPC requests.  Only one instance
+     * of Custom Endpoints is created for each instance of the Kinvey Client.
+     *
+     *
+     *
+     * This method is thread-safe.
+     *
+     *
+     *
+     * Sample Usage:
+     * <pre>
+     * `CustomEndpoints myCustomEndpoints = kinveyClient.customEndpoints();
+    ` *
+    </pre> *
+     *
+     *
+     * @return Instance of [com.kinvey.java.UserDiscovery] for the defined collection
+     */
+    @Deprecated("")
```

Comment:
let's add some useful message why this is now deprecated

Reply:
Added text to point user to use new method.